src/System.Private.CoreLib/shared/System/Numerics/Vector.cs

```diff
@@ -1409,108 +1408,108 @@ public Vector(Span<T> values)
             {
                 if (typeof(T) == typeof(byte))
                 {
-                    hash = HashHelpers.Combine(hash, this.register.byte_0.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_1.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_2.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_3.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_4.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_5.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_6.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_7.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_8.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_9.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_10.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_11.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_12.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_13.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_14.GetHashCode());
-                    hash = HashHelpers.Combine(hash, this.register.byte_15.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_0.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_1.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_2.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_3.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_4.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_5.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_6.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_7.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_8.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_9.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_10.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_11.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_12.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_13.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_14.GetHashCode());
+                    hash = HashCode.Combine(hash, this.register.byte_15.GetHashCode());
                     return hash;
```

Comment:
Why not the Combine overload with 8 arguments?

Reply:
I think it would be better to use the `long` or `int` path for all different integer type sizes.

Reply:
> Why not the Combine overload with 8 arguments?

This is a non-hardware-accelerated fallback path.  I didn't want to force a dependency onto these long generic Combine overloads that are otherwise unused.

> I think it would be better to use the long or int path for all different integer type sizes.

@jkotas, e.g.
```C#
                    hash = HashCode.Combine(
                        Unsafe.ReadUnaligned<int>(ref this.register.byte_0),
                        Unsafe.ReadUnaligned<int>(ref this.register.byte_4),
                        Unsafe.ReadUnaligned<int>(ref this.register.byte_8),
                        Unsafe.ReadUnaligned<int>(ref this.register.byte_12));
```
?

Reply:
Oh, nevermind, I see what you meant, just following the existing int/long code paths for the other types as well.